app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comment_entries, dependent: :destroy
+  has_many :like_entries, dependent: :destroy
+
+  def update_user_posts_counter
+    author.update(post_counter: author.posts.count)
+  end
```

Comment:
:red_circle: [REQUIRED] This method does not increment the `post_counter` attribute. The update method takes a hash as its argument, and the key-value pair `post_counter`: `author.posts.count` tells the update method to update the post_counter attribute to the current number of posts the author has created.

So, If the author has no posts, then the value of the post_counter attribute will be 0. This is because the posts.count method will return 0 if the author has no posts.

The better approach to increment the post_counter attribute by 1 every time a new post is added is to use the increment! method. The increment! method will increment the value of the attribute by 1, and it will also save the changes to the database.

Here is an example of how to use the increment! method to increment the post_counter attribute:

```
def update_user_posts_counter
  author.increment!(:posts_counter)
end
```
This code will first find the Author. Then, it will increment the post_counter attribute by 1. Finally, it will save the changes to the database.

You can call the increment!(:post_counter) method after save using an after_save callback. An after_save callback is a method that is called after an object is saved to the database.

So your final code will look like :arrow_down: 

```
after_save :update_user_posts_counter

  private

  def update_user_posts_counter
    author.increment!(:posts_counter)
  end

```

Reply:
@bienvenuushindi done